recipes/jsonnet/all/conanfile.py

```diff
@@ -0,0 +1,77 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class JsonnetConan(ConanFile):
+    name = "jsonnet"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/google/jsonnet"
+    description = "Jsonnet - The data templating language"
+    topics = ("config", "json", "functional", "configuration")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake", "cmake_find_package"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def configure(self):
+        if self.settings.compiler not in ["gcc", "clang", "apple-clang"]:
+            raise ConanInvalidConfiguration("{} compiler not supported"
+                                            .format(self.settings.compiler))
```

Comment:
```suggestion
        if self.settings.compiler not in ["gcc", "clang", "apple-clang"]:
            raise ConanInvalidConfiguration("{} compiler not supported"
                                            .format(self.settings.compiler))
```

Move it to validate(self) method.

Reply:
Fixed